number_theory/utility.h

```diff
@@ -0,0 +1,36 @@
+#ifndef NUMBER_THEORY_UTILITY_H_
+#define NUMBER_THEORY_UTILITY_H_
+
+#include <limits>
+#include <functional>
+
+// Public Utility functions.
+
+namespace number_theory
+{
+
+// Accumulates values according to the binary representation of |binary|.
+// For each bit of |binary|, from lower to higher, it calls |operation| to
```

Comment:
I think instead of saying it `calls |opreation|`, we could say it `applies |operation|`. In this case, `applies |function|` might be even clearer.

Reply:
I change to `applies |operation|`, because `function` is a too general name.